internal/cli/command/controlplane/debug.go

```diff
@@ -0,0 +1,241 @@
+	logResult("add pipeline/resources/banzaicloud/configmaps.txt", addFile(tarWriter, "pipeline/resources/banzaicloud/configmaps.txt", fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "configmaps", "-owide", "-nbanzaicloud"}, env))) // -owide does not include contents
+	logResult("add pipeline/resources/banzaicloud/secrets.txt", addFile(tarWriter, "pipeline/resources/banzaicloud/secrets.txt", fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "secrets", "-owide", "-nbanzaicloud"}, env)))          // -owide does not include contents
+	logResult("add pipeline/resources/banzaicloud/events.yaml", addFile(tarWriter, "pipeline/resources/banzaicloud/events.yaml", fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "events", "-oyaml", "-nbanzaicloud"}, env)))
+	// TODO add helm binary to installer image
+	logResult("add pipeline/resources/banzaicloud/helm_list.txt", addFile(tarWriter, "pipeline/resources/banzaicloud/helm_list.txt", fetchContainerCommandOutputAndError(options.cpContext, []string{"helm", "list", "--namespace", "banzaicloud", "--all"}, env)))
+
+	logResult("create pipeline/logs", addDir(tarWriter, "pipeline/logs"))
+	logResult("create pipeline/logs/banzaicloud", addDir(tarWriter, "pipeline/logs/banzaicloud"))
+	pods := strings.Split(strings.TrimSpace(fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "pods", "-oname", "-nbanzaicloud"}, env)), "\n")
+	for _, pod := range pods {
+		pod = strings.TrimPrefix(strings.TrimSpace(pod), "pod/")
+		logResult("add pod log", addFile(tarWriter, filepath.Join("pipeline/logs/banzaicloud/", pod+".log"), fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "logs", "-nbanzaicloud", pod, "--all-containers"}, env)))
+	}
+
+	log.Infof("debug bundle has been written to %q", path)
+	logResult("add meta.log", addFile(tarWriter, "meta.log", logBuffer.String()))
+
+	return nil
+}
+
+// simpleCommand runs the given shell command, and returns its outputs, an error message or both
+func simpleCommand(command string, args ...string) string {
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, "sh", "-c", command)
+	if len(args) > 0 {
+		cmd = exec.CommandContext(ctx, command, args...)
+	}
+	out, err := cmd.CombinedOutput()
+	output := strings.TrimSpace(string(out))
+	if err != nil {
+		if output != "" {
+			output += "\n"
+		}
+		output += err.Error()
+	}
+	return output
+}
+
+func logResult(what string, err error) {
+	if err == nil {
+		log.Debugf("%s succeeded", what)
+	} else {
+		log.Errorf("%s failed: %v", what, err)
+	}
+}
+
+func addDir(w *tar.Writer, name string) error {
+	name = strings.TrimRight(filepath.Join(basePath, name), "/") + "/"
+	err := w.WriteHeader(&tar.Header{Name: name, Mode: 0777})
```

Comment:
**Recommended**: i think this will trigger [GoSec G301: Poor file permissions used when creating a directory](https://github.com/securego/gosec/blob/f13b8bc63970f558528e8c21dbff530bbbd57c02/testutils/source.go#L1346), we **SHOULD** use `0o744`.

Same at
https://github.com/banzaicloud/banzai-cli/pull/297/files?file-filters%5B%5D=.go#diff-4d612959f08e84e921149cda21dcd60fR249

Reply:
0744 is not that meaningful for a directory, 0755 or 0700 may be considered. but normally I'd expect the user to set up umask according to their preferences. anyway, this is a tar archive, the final modes will be determined by the tar on the machine which extracts this.

Reply:
`0o700`is fine IMO.

Reply:
I let this be postponed until/if we use said linter.